1.oid4vci_test_deployment.sh

```diff
@@ -18,34 +18,210 @@ fi
+  jq --arg keystore "$KEYCLOAK_KEYSTORE_FILE" \
+  --arg keystorePassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+  --arg keystoreType "$KEYCLOAK_KEYSTORE_TYPE" \
+  --arg keyAlias "$KEYCLOAK_KEYSTORE_RSA_ENC_KEY_ALIAS" \
+  --arg keyPassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+  '.config.keystore = [$keystore] | 
+   .config.keystorePassword = [$keystorePassword] |
+   .config.keystoreType = [$keystoreType] | 
+   .config.keyAlias = [$keyAlias] | 
+   .config.keyPassword = [$keyPassword]' \
+  > $TARGET_DIR/encryption_key_rsa-tmp.json 
+
+# echo "Configuring hmac signature key provider..."
+# HMAC_SIG_KEY_ID=$(uuidgen)
+# less $WORK_DIR/signature_key_hmac.json | \
+#   jq --arg keystore "$KEYCLOAK_KEYSTORE_FILE" \
+#   --arg keystorePassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+#   --arg keystoreType "$KEYCLOAK_KEYSTORE_TYPE" \
+#   --arg keyAlias "$KEYCLOAK_KEYSTORE_HMAC_SIG_KEY_ALIAS" \
+#   --arg kid "$HMAC_SIG_KEY_ID" \
+#   --arg keyPassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+#   '.config.keystore = [$keystore] | 
+#    .config.keystorePassword = [$keystorePassword] |
+#    .config.keystoreType = [$keystoreType] | 
+#    .config.keyAlias = [$keyAlias] | 
+#    .config.kid = [$kid] | 
+#    .config.keyPassword = [$keyPassword]' \
+#   > $TARGET_DIR/signature_key_hmac-tmp.json 
+
+# echo "Configuring aes enc key provider..."
+# AES_ENC_KEY_ID=$(uuidgen)
+# less $WORK_DIR/encryption_key_aes.json | \
+#   jq --arg keystore "$KEYCLOAK_KEYSTORE_FILE" \
+#   --arg keystorePassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+#   --arg keystoreType "$KEYCLOAK_KEYSTORE_TYPE" \
+#   --arg keyAlias "$KEYCLOAK_KEYSTORE_AES_ENC_KEY_ALIAS" \
+#   --arg kid "$AES_ENC_KEY_ID" \
+#   --arg keyPassword "$KEYCLOAK_KEYSTORE_PASSWORD" \
+#   '.config.keystore = [$keystore] | 
+#    .config.keystorePassword = [$keystorePassword] |
+#    .config.keystoreType = [$keystoreType] | 
+#    .config.keyAlias = [$keyAlias] | 
+#    .config.kid = [$kid] | 
+#    .config.keyPassword = [$keyPassword]' \
+#   > $TARGET_DIR/encryption_key_aes-tmp.json 
 
-# Manually copy the content of your PEM file into issuer-key.json if you generate a new PEM file
 # Register the EC-key with Keycloak
-echo "Registering issuer key..."
-$KC_INSTALL_DIR/bin/kcadm.sh create components -r master -o -f - < $WORK_DIR/issuer_key.json || { echo 'Key registration failed' ; exit 1; }
+echo "Registering issuer key ecdsa..."
+$KC_INSTALL_DIR/bin/kcadm.sh create components -r master -o -f - < $TARGET_DIR/issuer_key_ecdsa-tmp.json || { echo 'ECDSA Issuer Key registration failed' ; exit 1; }
+echo "Registering issuer key rsa..."
+$KC_INSTALL_DIR/bin/kcadm.sh create components -r master -o -f - < $TARGET_DIR/issuer_key_rsa-tmp.json || { echo 'RSA Issuer Key registration failed' ; exit 1; }
+echo "Registering encryption key rsa..."
+$KC_INSTALL_DIR/bin/kcadm.sh create components -r master -o -f - < $TARGET_DIR/encryption_key_rsa-tmp.json || { echo 'RSA Encryption Key registration failed' ; exit 1; }
```

Comment:
I encountered an error during execution:

```
Failed to load keys. Keys for algorithm RSA-OAEP are not supported.
RSA Encryption Key registration failed
```

It seems Keycloak doesn't support RSA-OAEP. We might need to use a different encryption algorithm or check Keycloak configurations for enabling RSA-OAEP support.